src/ast/ast_validations.c

```diff
@@ -782,8 +782,21 @@ static AST_Validation _Validate_DELETE_Clauses(const AST *ast, char **reason) {
 	if(!delete_clause) return AST_VALID;
 
 	const cypher_astnode_t *match_clause = AST_GetClause(ast, CYPHER_AST_MATCH);
-	if(!match_clause) return AST_INVALID;
-
+	// Check for delete a projected entity such in "MATCH (n) WITH n AS x DELETE x".
+	const cypher_astnode_t *with_clause = AST_GetClause(ast, CYPHER_AST_WITH);
+	assert(!(with_clause && match_clause));
+	if(!match_clause && !with_clause) return AST_INVALID;
+
+	// Validate that every deleted object is an identifier and not property.
+	uint nitems = cypher_ast_delete_nexpressions(delete_clause);
+	for(uint i = 0; i < nitems; i++) {
+		const cypher_astnode_t *ast_expr = cypher_ast_delete_get_expression(delete_clause, i);
+		if(cypher_astnode_type(ast_expr) != CYPHER_AST_IDENTIFIER) {
+			asprintf(reason, "DELETE support the removal of valid graph entities only.");
```

Comment:
Is this the same error we're getting from Neo?

Reply:
Neo returns runtime error when deleting something other than a graph entity. As for the compile-time exception, it only alerts when the removed expression identifier appears for the first time, in the DELETE clause.